client/src/actions/rounds.ts

```diff
@@ -75,195 +67,118 @@ const loadingError = (address: string, error: string): RoundsActions => ({
-    dispatch(loadingError(address, "error loading round start time"));
-    console.error(e);
-    return;
-  }
-
-  dispatch({
-    type: ROUNDS_LOADING_ROUND,
-    address,
-    status: Status.LoadingRoundEndTime,
-  });
-
-  let roundEndTime;
-  try {
-    const ret: BigNumber = await contract.roundEndTime();
-    roundEndTime = ret.toNumber();
-  } catch (e) {
-    dispatch(loadingError(address, "error loading round end time"));
-    console.error(e);
-    return;
-  }
-
-  dispatch({
-    type: ROUNDS_LOADING_ROUND,
-    address,
-    status: Status.LoadingToken,
-  });
-
-  let token;
-  try {
-    token = await contract.token();
-  } catch (e) {
-    dispatch(loadingError(address, "error loading round token"));
-    console.error(e);
-    return;
-  }
-
-  dispatch({
-    type: ROUNDS_LOADING_ROUND,
-    address,
-    status: Status.LoadingRoundMetaPtr,
-  });
-
-  let roundMetaPtr: MetaPtr;
-  try {
-    roundMetaPtr = await contract.roundMetaPtr();
-  } catch (e) {
-    dispatch(loadingError(address, "error loading round metaPtr"));
-    console.error(e);
-    return;
-  }
-
-  dispatch({
-    type: ROUNDS_LOADING_ROUND,
-    address,
-    status: Status.LoadingRoundMetadata,
-  });
-
-  let roundMetadata: RoundMetadata;
-  try {
-    const resp = await pinataClient.fetchText(roundMetaPtr.pointer);
-    roundMetadata = JSON.parse(resp);
-  } catch (e) {
-    dispatch(loadingError(address, "error loading round metadata"));
-    console.error(e);
-    return;
-  }
-
-  dispatch({
-    type: ROUNDS_LOADING_ROUND,
-    address,
-    status: Status.LoadingApplicationMetaPtr,
-  });
-
-  let applicationMetaPtr: MetaPtr;
+export const getRoundApplicationMetadata = async (
+  pointer: string
+): Promise<RoundApplicationMetadata | null> => {
+  const pinataClient = new PinataClient();
   try {
-    applicationMetaPtr = await contract.applicationMetaPtr();
+    const resp = await pinataClient.fetchText(pointer);
+    return JSON.parse(resp) as RoundApplicationMetadata;
   } catch (e) {
-    dispatch(loadingError(address, "error loading application metaPtr"));
     console.error(e);
-    return;
+    return null;
   }
+};
 
-  dispatch({
-    type: ROUNDS_LOADING_ROUND,
-    address,
-    status: Status.LoadingApplicationMetadata,
-  });
-
-  let applicationMetadata: RoundApplicationMetadata;
-  let projectQuestionId;
-  let recipientQuestionId;
-  try {
-    const resp = await pinataClient.fetchText(applicationMetaPtr.pointer);
-    applicationMetadata = JSON.parse(resp);
-
-    projectQuestionId = applicationMetadata.applicationSchema.length;
-    applicationMetadata.applicationSchema.unshift({
-      ...projectQuestion,
-      id: projectQuestionId,
+export const loadRound =
+  (roundInfo: RoundResponse) => async (dispatch: Dispatch) => {
+    if (
+      !roundInfo ||
+      roundInfo.round.roundMetaPtr.pointer === undefined ||
+      roundInfo.round.applicationMetaPtr.pointer === undefined
+    ) {
+      // TODO: log error and metric
```

Comment:
shall we dispatch an error to be able to show it in the UI?

Reply:
> do we actually need the network switch? it's useful for development but in theory in production we should just stay on the network where the project registry is deployed /cc @michellema1208

I removed it from the header where the chain is displayed.

Reply:
Also set up env to only allow OP mainnet for prod for OP launch and our test chains for dev env

Reply:
this was meant for the previous comment...

Reply:
> shall we dispatch an error to be able to show it in the UI?

This is coming as well